File: Python/Program/Mahasiswa.py
# Import Class yang digunakan
from SivitasAkademik import SivitasAkademik
from ProgramStudi import ProgramStudi

class Mahasiswa(SivitasAkademik):
    # Kelas Mahasiswa di sini berperan sebagai:
    # - Child dari Kelas SivitasAkademik
    # - GrandChild dari Kelas Human
    
    # ATRIBUT
    __NIM = ""
    __semester = ""

    # METHODS
    # Constructor
    def __init__(self, nik="", nama="", jk='',  asal_univ="", email_edu = "", fakultas="", prodi = ProgramStudi(), nim = "", semester = ""):
        super(Mahasiswa, self).__init__(nik, nama, jk, asal_univ, email_edu, fakultas, prodi)
        self.__NIM = nim
        self.__semester = semester

    # Setter and Getter
    def getNIM(self):
        return self.__NIM
    
    def setNIM(self, nim):
        self.__NIM = nim

    def getSemester(self):
        return self.__semester
    
    def setSemester(self, sem):
        self.__semester = sem

File: CPP/Program/mahasiswa.cpp
#include "header.hh"

// == Kelas Mahasiswa ==

// Constructors
Mahasiswa::Mahasiswa()
{
    this->NIM = "";
    this->semester = 1;
}
Mahasiswa::Mahasiswa(string nim, int semester)
{
    this->NIM = nim;
    this->semester = semester;
}
Mahasiswa::Mahasiswa(string nik, string nama, char jk, string asal_universitas, string email_edu, string fakultas, ProgramStudi *prodi, string nim, int semester) : SivitasAkademik(nik, nama, jk, asal_universitas, email_edu, fakultas, prodi)
{
    this->NIM = nim;
    this->semester = semester;
}

// Setter and Getter
string Mahasiswa::getNIM()
{
    return this->NIM;
}
int Mahasiswa::getSemester()
{
    return this->semester;
}
void Mahasiswa::setNIM(string nim)
{
    this->NIM = nim;
}
void Mahasiswa::setSemester(int semester)
{
    this->semester = semester;
}

// Destructors
Mahasiswa::~Mahasiswa()
{
}